06_HelloGraphicsQueue/main.cpp

```diff
@@ -2,80 +2,436 @@
+
+		cmdbuf->end();
+
+		queue->startCapture();
+		queue->submit(1u, &submitInfo, done.get());
+		queue->endCapture();
+		m_device->blockForFences(1u, &done.get());
+
+		// Read the buffer back, create an ICPUImage with an adopted buffer over its contents.
+
+		// Map memory, so contents of `outputImageBuffer` will be host visible.
+		const IDeviceMemoryAllocation::MappedMemoryRange memoryRange(outputBufferAllocation.memory.get(), 0ull, outputBufferAllocation.memory->getAllocationSize());
+		auto imageBufferMemPtr = m_device->mapMemory(memoryRange, IDeviceMemoryAllocation::EMCAF_READ);
+		if (!imageBufferMemPtr)
+			return logFail("Failed to map the Device Memory!\n");
+
+		// If the mapping is not coherent the range needs to be invalidated to pull in new data for the CPU's caches.
+		if (!outputBufferAllocation.memory->getMemoryPropertyFlags().hasFlags(IDeviceMemoryAllocation::EMPF_HOST_COHERENT_BIT))
+			m_device->invalidateMappedMemoryRanges(1, &memoryRange);
+
+		const auto ouputImageCreationParams = smallImg->getCreationParameters();
+		// While JPG/PNG/BMP/EXR Loaders create ICPUImages because they cannot disambiguate colorspaces,
+		// 2D_ARRAY vs 2D and even sometimes formats (looking at your PNG normalmaps!),
+		// the writers are always meant to be fed by ICPUImageViews.
+		ICPUImageView::SCreationParams params = {};
+		{
+			params.image = ICPUImage::create(ouputImageCreationParams);
+			{
+				auto cpuView = core::make_smart_refctd_ptr<CDummyCPUBuffer>(smallImgByteSize, imageBufferMemPtr, core::adopt_memory_t());
+				ICPUImage::SBufferCopy region = {};
+				region.imageSubresource.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT;
+				region.imageSubresource.layerCount = 1;
+				region.imageExtent = ouputImageCreationParams.extent;
+				params.image->setBufferAndRegions(std::move(cpuView),core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1,region));
+			}
+			// Only DDS and KTX support exporting layered views.
+			params.viewType = ICPUImageView::ET_2D;
+			params.format = ouputImageCreationParams.format;
+			params.subresourceRange.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT;
+			params.subresourceRange.layerCount = 1;
 		}
+		auto cpuImageView = ICPUImageView::create(std::move(params));
+		asset::IAssetWriter::SAssetWriteParams writeParams(cpuImageView.get());
+		m_assetMgr->writeAsset("blit.png", writeParams);
+
+		m_device->unmapMemory(outputBufferAllocation.memory.get());
+
+		return true;
+	}
+
+	//
+	void workLoopBody() override {}
+
+	//
+	bool keepRunning() override {return false;}
+
+protected:
+	core::vector<queue_req_t> getQueueRequirements() const override
+	{
+		using flags_t = IPhysicalDevice::E_QUEUE_FLAGS;
+		return {{.requiredFlags=flags_t::EQF_GRAPHICS_BIT,.disallowedFlags=flags_t::EQF_NONE,.queueCount=1,.maxImageTransferGranularity={1,1,1}}};
```

Comment:
if you want a GRAPHICS+TRANSFER queue with `getQueue` then you MUST declare this here